File: app/src/main/java/com/zillennium/utswap/bases/websocket/WSAction.java
package com.zillennium.utswap.bases.websocket;

import okhttp3.WebSocket;
import okio.ByteString;
import rx.functions.Action1;

/**
 * please use {@link WS}
 */
public abstract class WSAction implements Action1<WSInfo> {

    @Override
    public void call(WSInfo webSocketInfo) {
        if (webSocketInfo.isOnOpen()) {

            onOpen(webSocketInfo.getWebSocket());
        } else {
            webSocketInfo.getString();
            onMessage(webSocketInfo.getString());
        }
    }

    public abstract void onOpen(WebSocket webSocket);

    public abstract void onMessage(String text);

    public abstract void onMessage(ByteString bytes);
}

File: app/src/main/java/com/zillennium/utswap/bases/websocket/WSAction.kt
package com.zillennium.utswap.bases.websocket

import okhttp3.WebSocket
import okio.ByteString
import rx.functions.Action1

/**
 * please use [WS]
 */
abstract class WSAction : Action1<WSInfo> {
    override fun call(webSocketInfo: WSInfo) {
        if (webSocketInfo.isOnOpen) {
            onOpen(webSocketInfo.webSocket)
        } else {
            webSocketInfo.string
            onMessage(webSocketInfo.string)
        }
    }

    abstract fun onOpen(webSocket: WebSocket?)
    abstract fun onMessage(text: String?)
    abstract fun onMessage(bytes: ByteString?)
}